src/features/locators/reducers/changeLocatorElement.thunk.ts

```diff
@@ -0,0 +1,68 @@
+import { ActionReducerMapBuilder, createAsyncThunk } from "@reduxjs/toolkit";
+import { LocatorTaskStatus, LocatorsState } from "../types/locator.types";
+import { locatorsAdapter, selectLocatorById } from "../locators.selectors";
+import { RootState } from "../../../app/store/store";
+import { ChangeLocatorAttributesPayload } from "../locators.slice";
+import { evaluateXpath } from "../utils/utils";
+import { generateId, getElementFullXpath, parseElementFromString } from "../../../common/utils/helpers";
+import { sendMessage } from "../../../pageServices/connector";
+
+type ChangeLocatorElementPayload = ChangeLocatorAttributesPayload;
+
+export const changeLocatorElement = createAsyncThunk(
+  "locators/changeLocatorElement",
+  async (payload: ChangeLocatorElementPayload, thunkAPI) => {
+    const { locator, element_id, ...rest } = payload;
+
+    let { foundHash } = JSON.parse(await evaluateXpath(locator, element_id));
+    const { foundElement } = JSON.parse(await evaluateXpath(locator, element_id));
+    const fullXpath = await getElementFullXpath(foundElement);
+    const parsedElement = parseElementFromString(foundElement);
+
+    const state = thunkAPI.getState() as RootState;
+
+    const _locator = selectLocatorById(state, element_id);
+
+    if (!_locator) return;
+
+    if (!foundHash) {
+      foundHash = generateId();
```

Comment:
Why do you generate new hash and don't use `element_id`?

Reply:
while editing we can move locator from one element to another. if with every edit we need to create a new hash, otherwise we'll have few nodes with same hashes in DOM.